### DISCUSSION TITLE
free host

### DISCUSSION THREAD

**Discussion by Xyvraaq** [jagrosh/MusicBot#1123] [Questions]
i know that free host are not recommended, but i need it, if someone knows a free host please tell me, thanks <3

**Comment by jagrosh** [comment]
Free hosts are not recommended, because at least one - if not all - of the following will apply:
- The quality will be subpar, the bot will be slow/laggy, or the bot may fail to load music entirely
- You won't have command-line access to the host to properly set up and run the bot
- The bot won't have permission to write files, or the host will not have file persistence, resulting in lost settings
- The service on which you are hosting isn't meant for round-the-clock applications like Discord bots, and you'll be breaking the ToS of the hosting service
- The host is malicious, and is only providing hosting in order to access the data you provide to it (albeit this is unlikely).

If you cannot pay for hosting and cannot run the bot on your home PC, then I would recommend finding an already-hosted music bot to use instead of attempting to use JMusicBot with all of these hurdles.
